### DISCUSSION TITLE
Size of size_t

### DISCUSSION THREAD

**Discussion by ot2t7** [mlua-rs/mlua#174] [Q&A] [answered]
Hey, 
When I compile a function with mlua it generates bytecode where the size of size_t is 8 bytes instead of the default 4 bytes. This is probably the intended behavior but I'm curious why the mlua dev has set it up like this. Also, is there any way I can change this?

**Comment by ot2t7** [comment]
mlua import: `mlua = { version = "0.8", features = ["lua51", "vendored"] }`
my rustc version: `rustc 1.61.0 (fe5b13d68 2022-05-18)`
my cargo version: `cargo 1.61.0 (a028ae4 2022-04-29)`

**Comment by ot2t7** [comment, marked as answer]
After further research, I've found that luac5.1 also generates bytecode where the size of size_t is 8 bytes. It seems because I'm on a 64 bit system it's making size_t be 64 bits? I guess that makes sense, I'll mark this discussion as answered.

**Comment by khvzak** [comment]
`sizeof(size_t)` is 8 bytes on x64, it's easy to proof:
```c
#include <stdio.h>

int main() {
    printf("%lu\n", sizeof(size_t));
}
```
prints `8`
